e2e-chatbot-app-next/packages/ai-sdk-providers/src/databricks-provider/responses-agent-language-model/responses-convert-to-input.ts

```diff
@@ -99,10 +104,54 @@ export async function convertToResponsesInput({
               break;
             }
             case 'tool-call': {
+              const toolName = providerOptions?.toolName ?? part.toolName;
```

Comment:
For my own understanding - is the reason we need extra logic here because Vercel's AI SDK doesn't yet understand or have a representation of tool call requests/approvals?